tensorboard/plugins/mesh_visualizer/mesh_summary.py

```diff
@@ -0,0 +1,180 @@
+
+  Args:
+    name: Uniquely identifiable name of the summary op. Could be replaced by
+      combination of name and type to make it unique even outside of this
+      summary.
+    display_name: Will be used as the display name in TensorBoard.
+      Defaults to `tag`.
+    description: A longform readable description of the summary data. Markdown
+      is supported.
+    tensor: Tensor to display in summary.
+    content_type: Type of content inside the Tensor.
+    json_config: A string, JSON-serialized dictionary of ThreeJS classes
+      configuration.
+    collections: List of collections to add this summary to.
+
+  Returns:
+    Tensor summary with metadata.
+  """
+  tensor = tf.convert_to_tensor(value=tensor)
+  tensor_metadata = metadata.create_summary_metadata(
+      name,
+      display_name,
+      content_type,
+      tensor.shape.as_list(),
+      description,
+      json_config=json_config)
+  tensor_summary = tf.summary.tensor_summary(
+      metadata.get_instance_name(name, content_type),
+      tensor,
+      summary_metadata=tensor_metadata,
+      collections=collections)
+  return tensor_summary
+
+
+def _get_display_name(name, display_name):
+  """Returns display_name from display_name and name."""
+  if display_name is None:
+    return name
+  return display_name
+
+
+def _get_json_config(config_dict):
+  """Parses and returns JSON string from python dictionary."""
+  json_config = '{}'
+  if config_dict is not None:
+    json_config = json.dumps(config_dict, sort_keys=True)
+  return json_config
+
+
+def op(name, vertices, faces=None, colors=None, display_name=None,
+       description=None, collections=None, config_dict=None):
+  """Creates a TensorFlow summary op for mesh rendering.
+
+  Args:
+    name: A name for this summary operation.
+    vertices: Tensor of shape `[dim_1, ..., dim_n, 3]` representing the 3D
+      coordinates of vertices.
+    faces: Tensor of shape `[dim_1, ..., dim_n, 3]` containing indices of
+      vertices within each triangle.
+    colors: Tensor of shape `[dim_1, ..., dim_n, 3]` containing colors for each
+      vertex.
+    display_name: If set, will be used as the display name in TensorBoard.
+      Defaults to `name`.
+    description: A longform readable description of the summary data. Markdown
+      is supported.
+    collections: Which TensorFlow graph collections to add the summary op to.
+      Defaults to `['summaries']`. Can usually be ignored.
+    config_dict: Dictionary with ThreeJS classes names and configuration.
+  Returns:
+    Merged summary for mesh/point cloud representation.
+  """
+  display_name = _get_display_name(name, display_name)
+  json_config = _get_json_config(config_dict)
+
+  # All tensors representing a single mesh will be represented as separate
+  # summaries internally. Those summaries will be regrouped on the client before
+  # rendering.
+  summaries = []
+  tensors = [(vertices, plugin_data_pb2.MeshPluginData.VERTEX),
```

Comment:
Suggest using a (4-space) hanging indent here instead of an aligned
indent. Aligned indents are brittle and require changes to entire blocks
of code when a variable is renamed. This problem disappears with hanging
indents:

```python
  tensors = [
      (vertices, plugin_data_pb2.MeshPluginData.VERTEX),
      (faces, plugin_data_pb2.MeshPluginData.FACE),
      (colors, plugin_data_pb2.MeshPluginData.COLOR),
  ]
```

Likewise below.